neuralmonkey/tf_manager.py

```diff
@@ -0,0 +1,178 @@
+"""Interface between the data and TF session.
+
+This module impelements the TensorFlow manager which encapsulates the graph
+execution in existing sessions.
+
+"""
+
+# pylint: disable=unused-import
+from typing import Any, List, Union
+
+import numpy as np
+import tensorflow as tf
+from neuralmonkey.logging import log
+from neuralmonkey.dataset import Dataset
+
+from neuralmonkey.runners.base_runner import ExecutionResult, \
+        RunResult, reduce_execution_results
+
+# tests: pylint,mypy
+
+
+class TensorFlowManager(object):
+    """Inteface between computational graph, data and TF sessions.
+
+    Attributes:
+        sessions: List of active Tensorflow sessions.
+    """
+
+    def __init__(self, num_sessions, num_threads, variable_files=None,
+                 gpu_allow_growth=True):
+        """Initialize a TensorflowManager.
+
+        At this moment the graph must already exist. This method initializes
+        required number of TensorFlow sessions and initializes them with
+        provided variable files if they are provided.
+
+        Args:
+            num_sessions: Number of sessions to be initialized.
+            num_threads: Number of threads sessions will run in.
+            variable_files: List of variable files.
+        """
+
+        session_cfg = tf.ConfigProto()
+        session_cfg.inter_op_parallelism_threads = num_threads
+        session_cfg.intra_op_parallelism_threads = num_threads
+        session_cfg.allow_soft_placement = True # needed for multiple GPUs
+        # pylint: disable=no-member
+        session_cfg.gpu_options.allow_growth = gpu_allow_growth
+
+        self.sessions = [tf.Session(config=session_cfg)
+                         for _ in range(num_sessions)]
+        init_op = tf.initialize_all_variables()
+        for sess in self.sessions:
+            sess.run(init_op)
+        self.saver = tf.train.Saver()
+
+        if variable_files:
+            if len(variable_files) != num_sessions:
+                raise Exception(("The number of provided variable files ({}) "
+                                 "is different than a number sessions ({})")
+                                .format(len(variable_files), num_sessions))
+            self.restore(variable_files)
+
+    # pylint: disable=too-many-locals
+    def execute(self,
+                dataset: Dataset,
+                execution_scripts,
```

Comment:
Co kdyby se předek runnerů a trainerů jmenoval `Executable` a to, co je teď  `Executable` by se jmenovala `ExePackage`?